test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,291 @@
+
+		ginkgo.By("Modifying PVC via VAC")
+		newPVC := SetPVCVACName(ctx, l.resource.Pvc, l.vac.Name, f.ClientSet, setVACWaitPeriod)
+		l.resource.Pvc = newPVC
+		gomega.Expect(l.resource.Pvc).NotTo(gomega.BeNil())
+
+		ginkgo.By("Waiting for modification to finish")
+		WaitForVolumeModification(ctx, l.resource.Pvc, f.ClientSet, modifyVolumeWaitPeriod)
+
+		pvcConditions := l.resource.Pvc.Status.Conditions
+		gomega.Expect(pvcConditions).To(gomega.BeEmpty(), "PVC should not have conditions")
+	})
+
+	ginkgo.It("should modify volume that already has a VAC", func(ctx context.Context) {
+		init(ctx, true /* volume created with VAC */)
+		ginkgo.DeferCleanup(cleanup)
+
+		vacDriver, _ := driver.(storageframework.VolumeAttributesClassTestDriver)
+		newVAC := vacDriver.GetVolumeAttributesClass(ctx, l.config)
+		gomega.Expect(newVAC).NotTo(gomega.BeNil())
+		_, err := f.ClientSet.StorageV1alpha1().VolumeAttributesClasses().Create(ctx, newVAC, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "While creating new VolumeAttributesClass")
+		ginkgo.DeferCleanup(CleanupVAC, newVAC, f.ClientSet, vacCleanupWaitPeriod)
+
+		ginkgo.By("Creating a pod with dynamically provisioned volume")
+		podConfig := e2epod.Config{
+			NS:            f.Namespace.Name,
+			PVCs:          []*v1.PersistentVolumeClaim{l.resource.Pvc},
+			SeLinuxLabel:  e2epod.GetLinuxLabel(),
+			NodeSelection: l.config.ClientNodeSelection,
+			ImageID:       e2epod.GetDefaultTestImageID(),
+		}
+		pod, err := e2epod.CreateSecPodWithNodeSelection(ctx, f.ClientSet, &podConfig, f.Timeouts.PodStart)
+		ginkgo.DeferCleanup(e2epod.DeletePodWithWait, f.ClientSet, pod)
+		framework.ExpectNoError(err, "While creating pod for modifying")
+
+		ginkgo.By("Modifying PVC via VAC")
+		newPVC := SetPVCVACName(ctx, l.resource.Pvc, newVAC.Name, f.ClientSet, setVACWaitPeriod)
+		l.resource.Pvc = newPVC
+		gomega.Expect(l.resource.Pvc).NotTo(gomega.BeNil())
+
+		ginkgo.By("Waiting for modification to finish")
+		WaitForVolumeModification(ctx, l.resource.Pvc, f.ClientSet, modifyVolumeWaitPeriod)
+
+		pvcConditions := l.resource.Pvc.Status.Conditions
+		gomega.Expect(pvcConditions).To(gomega.BeEmpty(), "PVC should not have conditions")
+	})
+}
+
+// SetPVCVACName sets the VolumeAttributesClassName on a PVC object
+func SetPVCVACName(ctx context.Context, origPVC *v1.PersistentVolumeClaim, name string, c clientset.Interface, timeout time.Duration) *v1.PersistentVolumeClaim {
+	pvcName := origPVC.Name
+	var updatedPVC *v1.PersistentVolumeClaim
+
+	gomega.Eventually(ctx, func(g gomega.Gomega) {
+		var err error
+		// Get existing PVC
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get(ctx, pvcName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "While getting existing PVC")
+
+		// Update PVC
+		updatedPVC.Spec.VolumeAttributesClassName = &name
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(ctx, updatedPVC, metav1.UpdateOptions{})
```

Comment:
you may want to do a patch instead of update. Update will fail if another field in the also gets updated in the meantime.

Reply:
Switched to patch